### PR TITLE
STRIP implementation

### LINKED ISSUE #41: [FEATURE] Implement core STRIP entropy scoring utility (no CLI wiring)
## Summary

Implement the core STRIP logic as an internal utility that can compute prediction entropies for perturbed inputs, but do not wire it into the CLI yet. This gives us a tested building block for later integration.

## Acceptance Criteria

- [x] A new module mithridatium/defenses/strip.py exists.

- [ ]There is a function that computes per-sample entropy from model logits.

- [x] There is a function that, given a model and a dataloader, computes STRIP-style entropy scores by:
	•      Selecting a small set of base samples from the dataset.
	•	Generating perturbed variants by mixing base inputs with randomly sampled other inputs.
	•	Running the model on these perturbed inputs and aggregating prediction entropies per base sample.

- [ ] All computations respect the model’s device and use float32, similar to the recent MMBD fix.

## Tasks that need to be completed for this feature

- Create mithridatium/defenses/strip.py with:
	•	def prediction_entropy(logits: torch.Tensor) -> torch.Tensor
                 Returns per-sample entropy over the softmax distribution.
	•	def strip_scores(model, dataloader, num_bases: int = 32, num_perturbations: int = 16, device=None) -> Dict[str, Any]
                 Returns raw entropy scores for each base sample (e.g., list/array of entropies).

- Inside strip_scores:
	•	Determine device from model.parameters() if not provided, falling back to get_device(0) if needed (same pattern as run_mmbd).
	•	Move model and inputs to the chosen device with dtype=torch.float32.
	•	Sample num_bases examples from the dataloader as “base” inputs.
	•	For each base input, create num_perturbations mixed images by combining it with randomly drawn other images (e.g., simple linear blend like 0.5 * base + 0.5 * other as a first pass).
	•	Run the model on all perturbed images, compute logits, then compute entropy using prediction_entropy.
	•	Aggregate entropy per base (e.g., mean entropy over its perturbed copies).

## Additional Context

This issue is just about the core math and device handling, not CLI or JSON reports.

Later issues will hook this into mithridatium detect and define how we expose results to users.

### PR DESCRIPTION
**Pull request by GustavoLucca**
## Description

Fixes #41

Began implementation of STRIP with key functions that calculate entropy for input images.

# 1. What was Changed

Created strip.py, with functions prediction_entropy that calculates entropy per sample image, and strip_scores which actually generates the entropy score to determine if a model is backdoored or not across multiple samples.

# 2. Why it was Changed

We are implementing a new defense to have a more rigid and robust framework for detecting backdoors in AI models.

# 3. How it was Changed

After creating the basis for strip.py, I decided to pursue excellence by creating test scripts which test both of the functions implemented in strip.py sequentially and separately. These returned passing tests, ensuring my implementation worked. All that is left is extending this implementation to the CLI.

### CODE REVIEW

File: mithridatium/defenses/strip.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import torch
```
Comment (pelumitegbe) [inline mithridatium/defenses/strip.py]: Nice work on the core STRIP implementation. The prediction_entropy and strip_scores functions match the intent of the issue and give us a clean, testable building block for the CLI integration.

**Review by pelumitegbe — APPROVED** [review]
Overall, great work
